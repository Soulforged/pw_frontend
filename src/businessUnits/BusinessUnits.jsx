//@flow
import React from "react";
import { CRUD, CRUDFilter, ActiveCell } from "src/components";

type Props = {
  openForm: () => void,
  fetchBusinessUnits: (Object) => void,
  businessUnits: Object,
  showDetails: (id: number) => void
};

const columns = [
  { Header: "ID", accessor: "id" },
  { Header: "Name", accessor: "name" },
  { Header: "Company Type", accessor: "companyType" },
  { Header: "Status", accessor: "status", Cell: ActiveCell },
];

const Filter = ({ fetchBusinessUnits }: { fetchBusinessUnits: (Object) => void }) => (
  <CRUDFilter filter={fetchBusinessUnits}>
    <div className="col-md-11 col-sm-11 txt-div">
      <label htmlFor="filter" className="lbl">
        Search text
        <input name="name" className="form-control" placeholder="Text to search" />
      </label>
    </div>
  </CRUDFilter>
);

export default (props: Props) => (
  <CRUD
    title="Business Unit List"
    createButtonTitle="New Business Unit"
    openForm={props.openForm}
    filter={<Filter {...props} />}
    columns={columns}
    showDetails={props.showDetails}
    entities={props.businessUnits}
    loader={props.fetchBusinessUnits}
  />
);
