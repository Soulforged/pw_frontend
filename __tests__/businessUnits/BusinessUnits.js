//@flow
import React from "react";
import BusinessUnits from "src/businessUnits/BusinessUnits";
import { shallow } from "enzyme";

it("matches snapshot", () => {
  const render = shallow(<BusinessUnits />).dive();
  expect(render).toMatchSnapshot();
});
