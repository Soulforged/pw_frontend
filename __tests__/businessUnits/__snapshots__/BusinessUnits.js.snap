// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <Filter />
  <div
    className="pnls"
    id="main-pnl"
  >
    <div>
      <h4
        className="form-header trebuchet bold"
      >
        Business Unit List
        <button
          className="add-new-btn pointer pull-right bold"
        >
          <i
            className="fa fa-plus theme"
          />
          New Business Unit
        </button>
      </h4>
      <div
        className="tbl-wrapper"
      >
        <lifecycle(branch(Component))
          columns={
            Array [
              Object {
                "Header": "ID",
                "accessor": "id",
              },
              Object {
                "Header": "Name",
                "accessor": "name",
              },
              Object {
                "Header": "Company Type",
                "accessor": "companyType",
              },
              Object {
                "Cell": [Function],
                "Header": "Status",
                "accessor": "status",
              },
            ]
          }
          createButtonTitle="New Business Unit"
          filter={<Filter />}
          title="Business Unit List"
        />
      </div>
    </div>
  </div>
</div>
`;
